src/main/java/com/revature/studyforce/notification/service/StacktraceSubscriptionService.java

```diff
@@ -0,0 +1,136 @@
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Service Layer for StacktraceSubscriptions {@link StacktraceSubscriptionRepository}
+ * @author Brandon Pinkerton
+ */
+@Service
+public class StacktraceSubscriptionService {
+
+    private StacktraceSubscriptionRepository stacktraceSubscriptionRepository;
+    private SubscriptionService subscriptionService;
+    private StacktraceRepository stacktraceRepository;
+
+
+
+    @Autowired
+    public StacktraceSubscriptionService(StacktraceSubscriptionRepository stacktraceSubscriptionRepository,
+                                         SubscriptionService subscriptionService,
+                                         StacktraceRepository stacktraceRepository){
+        this.stacktraceSubscriptionRepository = stacktraceSubscriptionRepository;
+        this.subscriptionService = subscriptionService;
+        this.stacktraceRepository = stacktraceRepository;
+    }
+
+
+    /**
+     * Retrieves a stacktraceSubscription by composite of stacktraceId and userId
+     * from {@link StacktraceSubscriptionRepository#findAllByStacktraceSubscriptionId(StacktraceSubscriptionId)}
+     * @param stacktraceId The stacktraceId of the flashcard to be retrieved.
+     * @param userId The userId of the subscription to be retrieved.
+     * @return The stacktrace subscription of that stacktrace for that user.
+     */
+    public StacktraceSubscription getStacktraceSubscriptionByStacktraceAndUserId(Integer stacktraceId, Integer userId){
+        return stacktraceSubscriptionRepository.findAllByStacktraceSubscriptionId(
+          new StacktraceSubscriptionId(
+                  Objects.requireNonNull(stacktraceRepository.findById(stacktraceId).orElse(null)).getStacktraceId(),
+                  subscriptionService.getSubscriptionByUserId(userId).getId()
+          )
+        );
+    }
+
+    /**
+     * Retrieves all stacktrace subscribers for a given stacktrace {@link StacktraceSubscriptionRepository#findAllByStacktrace_StacktraceId(Integer)}
+     * @param stacktraceId The stacktraceId of the stacktrace to retrieve all of the subscriptions to it
+     * @return A list of all stacktrace subscriptions to a specific stacktrace
+     */
+    public List<StacktraceSubscription> getAllSubscribersByStacktraceId(Integer stacktraceId){
+        return stacktraceSubscriptionRepository.findAllByStacktrace_StacktraceId(stacktraceId);
+    }
+
+    /**
+     * Retrieves all stacktraces subscriptions for a given user {@link StacktraceSubscriptionRepository#findAllBySubscription_User_UserId(Integer)}
+     * @param userId The userId of the user to retrieve all of their subscriptions
+     * @return A list of all stacktraces subscriptions to a specific user
+     */
+    public List<StacktraceSubscription> getAllSubscriptionsByUserId(Integer userId){
+        return stacktraceSubscriptionRepository.findAllBySubscription_User_UserId(userId);
+    }
```

Comment:
Refactor with pagination

Reply:
This is an internal service method, not something to be hit by an endpoint and served in the frontend.